BUGFIX: add nanmean to zlp alignment, works for lazy signals

to266 opened the pull request:
Fixes https://github.com/hyperspy/hyperspy/issues/1592

TODO:
 - [x] tests

woozey committed:
Merge pull request #23 from hyperspy/RELEASE_next_minor

Merge recent changes from hs

jan-car committed:
Added desktop.ini to ignore list.

to266 committed:
slicing lazily loaded models

thomasaarholt committed:
Fix and add test

to266 committed:
BUGFIX: add nanmean to zlp alignment, works for lazy signals

to266 committed:
removed without_nans, as it was not used anywhere

to266 committed:
pep8

jan-car committed:
Merge remote-tracking branch 'origin/RELEASE_next_patch' into RELEASE_next_patch

francisco-dlp committed:
Merge pull request #1663 from thomasaarholt/fix1661

Fix fit_component multifit and add test

jan-car committed:
Merge remote-tracking branch 'origin/RELEASE_next_patch' into RELEASE_next_patch

francisco-dlp committed:
Merge pull request #1673 from CodeMonkeyJan/RELEASE_next_patch

Release next patch

ericpre committed:
replace deprecated method

ericpre committed:
fix test_eels

ericpre committed:
set matplotlib figure.max_open_warning in conftest.py to avoid warning

ericpre committed:
move Test1D.test_integrate1D to test_tools.py

ericpre committed:
set dask=0.14.3 to see if it is fixed the tests failing since recently.

ericpre committed:
fix bss test failing with numpy 1.13: the test was using np.diff on boolean array.

ericpre committed:
Fix test failure for sympy 1.1 release. Just `sympify` the string does the job. See #1686 and https://github.com/sympy/sympy/issues/12597 for details.

ericpre committed:
Fix issue with reading stack (FEI ser)

ericpre committed:
set dask=0.14.3 to see if it is fixed the tests failing since recently. (reverted from commit ca67b0fc8efb7fea347af05388aa7d68efdab964)

to266 committed:
comply with dask 0.15

ericpre committed:
Merge pull request #5 from to266/dask_axis_indexing

comply with dask 0.15

jat255 committed:
switch to integer division for label detection

francisco-dlp committed:
Merge pull request #1692 from jat255/FIX_plot_label_bug

FIX: switch to integer division for plot_images label detection

ericpre committed:
- fix bss test failing with numpy 1.13: the test was using np.diff on boolean array. (reverted from commit d333c57727a64fcfc00c5ba0224b1dd70f9c232f)
- exclude numpy=1.13.0

to266 committed:
Merge pull request #1675 from ericpre/fix_deprecation_warning

Fix tests failing after a few new dependency releases

jat255 committed:
Add code to reverse order of legend labeling when using 'cascade' or 'overlap' styles in plot_spectra

jat255 committed:
Update plot files for tests

jat255 committed:
Put legend reversing code into internal function

ericpre committed:
Merge pull request #1695 from jat255/ENH_plot_spectra_legend_order

Reverse plot_spectra legend order

jan-car committed:
Added preferences.GUIs.warn_if_guis_are_missing option, to
toggle warnings, if hyperspy_gui_ipywidgets or hyperspy_gui_traitsui
are missing.

jan-car committed:
Added comment about turning off warnings in the user guide.

jan-car committed:
Added small codeblock to show how to turn off warnings.

jan-car committed:
Small rephrasing.

ericpre committed:
Merge pull request #1701 from CodeMonkeyJan/make_gui_warning_optional

Make warnings about missing GUI features optional

ericpre committed:
Trigger events only once, when indices or coordinates are set. Don't trigger events if they are no changes.

woozey committed:
Merge pull request #24 from hyperspy/RELEASE_next_patch

Merge changes from HS:RNP

woozey committed:
Changes in setup.py to fix issue 1671

to266 committed:
Merge pull request #1709 from woozey/fix_1671

BUGFIX: Added holography example signals to setup.py (#1671)

askorikov committed:
Update mrc.py

francisco-dlp committed:
Merge branch 'RELEASE_next_patch' of https://github.com/hyperspy/hyperspy into RELEASE_next_patch

ericpre committed:
Add shell=True argument to subprocess.Popen to fix install dev version in anaconda on windows.

ericpre committed:
Fix and tidy up path in setup.py.

ericpre committed:
Fix typo `test_require` instead of `tests_require` in setup.py.

sem-geologist committed:
add error resolution

for bcf TraitError

sem-geologist committed:
add function to change 

decimal and thousand separator
for eventual hack around non-compliant XML of Bruker

sem-geologist committed:
Update bcf.py

sem-geologist committed:
Update io.py

remove else --- always (re-raise) the exception

ericpre committed:
Fix appveyor install pip.

ericpre committed:
Fix install pip in appveyor take2.

ericpre committed:
attempt without upgrading pip.

sem-geologist committed:
revert back io.py

no need for exception intercept,
automatic decimal point search and replace solves the issue,
no interaction from user is needed

sem-geologist committed:
make automatic check for

comma as decimal point in XML values.
No more user interaction is needed

sem-geologist committed:
add decimal point test to test_bcf.py

test if the general regular expression (defined in bcf.py)
is correctly catching the numeric values in xml with
comma as decimal separator

sem-geologist committed:
add short comment on 

defined regular expression

sem-geologist committed:
clean-up

sem-geologist committed:
fix str to byte str

sem-geologist committed:
fix example xml closing tags in the test

sem-geologist committed:
add: recognition of
scientific notation (e.g. 8,843E-8)

sem-geologist committed:
update regex explanation

Update regex:
add escape `\` before `\d`
to get ready for future deprecation

francisco-dlp committed:
Merge pull request #1744 from ericpre/fix_appveyor

Fix appveyor

sem-geologist committed:
Update test_bcf.py

ericpre committed:
Merge pull request #1741 from sem-geologist/patch-1

Add hack for BCF used MSXML decimal non-compliance

thomasaarholt committed:
Incorrect docs for format of msa saving

ericpre committed:
Add test for setting indices of the `axes_manager` and count the number of callback calls.

ericpre committed:
Use mock library instead of making a count class. Add tests for setting coordinates.

to266 committed:
Merge pull request #1710 from ericpre/improve_triggering_event_when_setting_axes_manager_indices

Trigger events only once when settings axes manager indices

ericpre committed:
Merge pull request #1762 from thomasaarholt/patch-2

Incorrect docs for format of msa saving

ericpre committed:
Add `optimize` argument to dm reader.

ericpre committed:
Change logger level for the `optimize` warning.

ericpre committed:
Update documentation on `optimize` argument of the dm reader.

jat255 committed:
enable reading of older ipr and spc files (and by extension, spd) from EDAX TEAM/vision software

jat255 committed:
Update edax reader to properly handle old versions of ipr and spc automatically

jat255 committed:
change file version comparison to "greater than"

jat255 committed:
add EDAX tests for old file versions, as well as loading all spc metadata

jat255 committed:
correct capitalization of file extensions

ericpre committed:
Merge pull request #1782 from jat255/BUGFIX_edax_old_version

Enable reading of older EDAX TEAM/Vision .spd and .spc files

jat255 committed:
Replace fixtures with module and class setup/teardown methods to fix broken Windows test

jat255 committed:
merge updated RELEASE_next_patch with improved EDAX compatability

jat255 committed:
clean up PEP8; remove some unneeded imports/attributes; correct some file names

francisco-dlp committed:
Merge pull request #1788 from jat255/BUGFIX_edax_pytest_errors

Fix for Pytest fixture errors

magnunor committed:
DOC: fix several minor typos in user_guide/tools.rst

francisco-dlp committed:
Merge pull request #1792 from magnunor/TYPO_tools_rst_RNP

DOC: fix several minor typos in user_guide/tools.rst (resubmission)

ericpre committed:
Fix np.issubdtype warning.

ericpre committed:
Use np.frombuffer instead of np.fromstring when appropriate.

ericpre committed:
Fix encoding in np.genfromtxt.

ericpre committed:
Fix default rcond in np.linalg.lstsq.

jat255 committed:
Various documentation fixes; shorten long lines, clean up code samples, update user guide images, fix typos, add gifs for some functions, etc.

jat255 committed:
Lena-be-gone

ericpre committed:
Fix issue with accessing multiple fields array following changes in numpy 1.14.

jat255 committed:
shorten up some more lines

jat255 committed:
Remove non-existent files from index.rst

jat255 committed:
Update bibliography to be nicer formatted; update citations throughout user guide to match

jat255 committed:
Rename CHANGES.txt to CHANGES.rst; Edit levels so detailed versions are hidden from TOC in user guide

jat255 committed:
Correct typos and  fix some long lines

francisco-dlp committed:
Merge pull request #1799 from ericpre/fix_deprecation_warning_and_issue_with_numpy1.14

Fix deprecation warning and issue with numpy1.14

ericpre committed:
Merge pull request #1798 from jat255/DOCFIX_update_user_guide

Update user guide, fix #1769, #1715 and #1794.

ericpre committed:
Merge upstream.

ericpre committed:
Improve docstring following review comments.

ericpre committed:
Fix one missed `np.fromstring()` to `np.frombuffer()` in bcf.py.

francisco-dlp committed:
Merge pull request #1723 from ericpre/fix_issue_install_dev_version_windows

Fix issue with installing the dev version in a windows anaconda

francisco-dlp committed:
Merge pull request #1806 from ericpre/fix_deprecation_warning_numpy1.14_take2

Fix deprecation warning and issue with numpy1.14 -take 2.

ericpre committed:
Revert back the log argument in `_make_sure_data_is_contiguous`.

francisco-dlp committed:
Fix typo

francisco-dlp committed:
Merge pull request #1781 from ericpre/fix_optimized_copy_warning

Fix #1360: optimized copy warning

francisco-dlp committed:
Merge remote-tracking branch 'upstream/RELEASE_next_patch' into lazy_improvements

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Add align_zlp test with masking (not passing)

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Remove unncessary da.nanmean

francisco-dlp committed:
Workaround for the failing test

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Don't compute the shift when shift == 0

francisco-dlp commented:
I have submitted added a test (that highlights a bug of course ;) ) to to266/hyperspy#20

francisco-dlp committed:
Automatic style corrections courtesy of autopep8

francisco-dlp committed:
Fix bug that highlights another one

align_zero_loss_peak was not taking into accout `mask` when `subpixel=True`

However, after correctly passing the kwarg to align1D a new `_map_iterate`
bug appears.

to266 committed:
Merge branch 'francisco-dlp-lazy_improvements' into lazy_improvements

to266 committed:
Work around np.nan not having shape (while other numpy floats do)

francisco-dlp commented:
I've just noticed that this was sent to the wrong branch (Rnm), I've merged it manually in Rnp beac4a6777a783b2909108dedde0a8376900c4a3
